spring-boot-project/spring-boot-docs/src/docs/antora/modules/reference/pages/testing/testcontainers.adoc

```diff
@@ -5,14 +5,69 @@ The https://www.testcontainers.org/[Testcontainers] library provides a way to ma
 In most cases, you will need to configure the application to connect to the service running in the container.
 
+In this case the lifecycle of the container instance is managed by Testcontainers framework, as described in official documentation.
+
+== Using via Spring managed beans
+
+The containers provided by Testcontainers framework can be managed by Spring Boot as beans.
+This method is often used in combination with javadoc:org.springframework.boot.testcontainers.service.connection.ServiceConnection[format=annotation].
+
+To use Testcontainer contains as Spring beans we need to create a configuration class declaring the container as bean:
+
+include-code::beandeclaration/BeanDeclarationConfig[]
+
+then we can start the container by importing the configuration class in the test class:
+
+include-code::beandeclaration/SpringTest[]
+
+
+== Using via importing container declaration classes
+
+A common pattern with Testcontainers framework is to declare the Container instances as static fields in an interface
+For example the following interface `MyInterface` declares two containers, one named `mongo` of type MongoDB and another named `neo` of type Neo4j:
+
+include-code::importcontainers/MyInterface[]
+
+When you have containers declared in this way, then you can have these containers managed by Spring Boot as beans.
+All that is needed to do that is adding javadoc:org.springframework.boot.testcontainers.context.ImportTestcontainers[format=annotation] to your configuration class as in:
+
+include-code::importcontainers/MyConfiguration[]
+
+== Lifecycle of managed containers
+
+If you have used the annotations and extensions provided by Testcontainers framework, then the lifecycle of container instances is managed by the Testcontainers framework.
+Please refer to the official documentation for the information about lifecycle of the containers, when managed by the Testcontainers framework.
+
+When the containers are managed by Spring as beans, then the lifecycle is clearly defined by Spring.
+The container beans are created and started before the beans of other types are created.
+This process ensures that any beans, which rely on functionality provided by the containers, can use those functionalities.
+
+The test containers can be started multiple times.
+Like any other beans the test containers are created and started once per application context managed by the TestContext Framework.
+For details about how TestContext framework manages the underlying application contexts and beans therein, please refer to the official Spring documentation.
+
+The container beans are stopped after the destruction of beans of other types.
+This ensures that any beans depending on the functionalities provided by the containers are cleaned up first.
+
+The containers are stopped as part of the application shutdown process, managed by the TestContext framework.
```

Comment:
This part is important to highlight. In some cases, for example, using activemq, kafka, pulsar and other containers, I recommend to declare it as a beans rather than static fields because that way the application context will shutdown the client first and then the container. Otherwise, when the container shutdown first the client will throw some errors.

Reply:
I understand that you would like to have a concrete example in addition to the general explanation already provided, so that the readers can more clearly understand the "why" behind usage. Is my understanding correct?

Reply:
Added as tips and notes.